### DISCUSSION TITLE
Restore backup over existing database but keep permissons

### DISCUSSION THREAD

**Discussion by tomaustin700** [dataplat/dbatools#8527] [Q&A] [answered]
Hi,
I'm wanting to restore the backup of one database over another database but keep the permissions and not overwrite them. It seems `Restore-DbaDatabase` will not allow me to do this, is this possible at all? Maybe using a combination of other commands. My thought process was to get all the security permissions of the database to be overwritten before I overwrite it and then reapply them after the restore but can't see an easy way to do this.

Any help would be appreciated.
Thanks

**Comment by ClaudioESSilva** [comment, marked as answer]
Hi @tomaustin700,

A database backup on SQL Server contains all permissions too. That means when you overwrite an existing database you will also overwrite the existing permissions with the ones that your backup brings.

Therefore that's nothing `Restore-DbaDatabase` can help with.

However, as you mentioned, on way to accomplish this process will be:
1- Script existing permissions (`Export-DbaUser`)
2- Restore the database (`Restore-DbaDatabase`)
3- Apply permissions exported on step 1 (`Invoke-DbaQuery`)

To give some guidance, take a look at this blog post [Replicate permissions to new Login/User with dbatools](https://claudioessilva.eu/2020/05/28/Replicate-permissions-to-new-Login/User-with-dbatools/ ) and all the other ones linked there (like the one that @SQLDBAWithABeard wrote).

Let us know if this helps.

**Reply by tomaustin700** [reply]
Thanks @ClaudioESSilva, that's what I suspected but thank-you for confirming
